ui/src/app/subnets/views/SpaceDetails/SpaceDetailsHeader/SpaceDelete/SpaceDelete.tsx

```diff
@@ -0,0 +1,51 @@
+import { Button, Strip, Notification } from "@canonical/react-components";
+import { useDispatch } from "react-redux";
+import { useHistory } from "react-router";
+
+import { actions as spaceActions } from "app/store/space";
+import type { Space } from "app/store/space/types";
+import { getCanBeDeleted } from "app/store/space/utils";
+import { getNetworksLocation } from "app/subnets/urls";
+
+type SpaceDeleteProps = {
+  space: Space;
+  handleClose?: () => void;
+};
+
+const SpaceDelete = ({ space, handleClose }: SpaceDeleteProps): JSX.Element => {
+  const canBeDeleted = getCanBeDeleted(space);
+  const dispatch = useDispatch();
+  const history = useHistory();
+  const handleDelete = () => {
+    dispatch(spaceActions.delete(space.id));
+    history.replace(getNetworksLocation({ by: "space" }));
```

Comment:
It would be good if this could check for (and display) errors before redirecting.